de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/progress/ConsoleProgressIndicator.java

```diff
@@ -0,0 +1,113 @@
+package de.fu_berlin.inf.dpp.server.progress;
+
+import de.fu_berlin.inf.dpp.monitoring.IProgressMonitor;
+
+import java.io.PrintStream;
+
+/**
+ * Progress indicator supposed to be use for terminal output.
+ *
+ * Refreshes the same line over and it over again with the current
+ * progress until the task is done to avoid flooding the output, but
+ * still be up-to-date.
+ *
+ * Redraws in the next line if stdout is used in the meantime
+ * (e.g. by a logger). Avoid excessive logging while this is in use.
+ *
+ * Alternatively use {@link de.fu_berlin.inf.dpp.monitoring.NullProgressMonitor}
+ * to disable progress reporting.
+ */
+public class ConsoleProgressIndicator implements IProgressMonitor {
+    private PrintStream out;
+    private boolean canceled = false;
+    
+    private int total = 0;
+    private int worked = 0;
+
+    private String task;
+    private String subTask;
+
+    private int animationFrame = 0;
+
+    public ConsoleProgressIndicator(PrintStream out) {
+        this.out = out;
+    }
+
+    @Override
+    public void done() {
+        out.print('\n');
+    }
+
+    @Override
+    public void subTask(String name) {
+        if (!name.equals(subTask)) {
+            subTask = name;
+            print();
+        }
+    }
+
+    @Override
+    public void setTaskName(String name) {
+        if (!name.equals(task)) {
+            task = name;
+            print();
+        }
+    }
+
+    @Override
+    public void worked(int amount) {
+        if (amount != worked) {
+            worked = amount;
+            print();
+        }
+    }
+
+    @Override
+    public void setCanceled(boolean canceled) {
+        if (canceled != this.canceled) {
+            this.canceled = canceled;
+            print();
+        }
+    }
+
+    @Override
+    public boolean isCanceled() {
+        return canceled;
+    }
+
+    @Override
+    public void beginTask(String name, int size) {
+        task = name;
+        total = size;
+        print();
+    }
+
+    private void print() {
+        if (canceled) {
+            out.print("[CANCELED] ");
+        } else if (total == IProgressMonitor.UNKNOWN) {
+            switch (animationFrame) {
+            case 0:
```

Comment:
Why not using a Array of strings in order to avoid the big switch:
```java
private String[] animationFrames= {"[-] ", "[\\] ", "[|] ", "[/] "};
...
out.print(animationFrames[animationFrameId]);
animationFrameId = (animationFrameId + 1) % animationFrames.length;
```
```

Reply:
Good suggestion. Fixed.